message_ix_models/model/water/data/water_for_ppl.py

```diff
@@ -85,26 +96,36 @@ def shares(
     corresponding cooling fraction
     """
     for col in search_cols_cooling_fraction:
-        # MAPPING ISOCODE to region name, assume one country only
         col2 = context.map_ISO_c[col] if context.type_reg == "country" else col
+
+        # Filter the cooling fraction
         cooling_fraction = hold_df[
             (hold_df["node_loc"] == col2)
             & (hold_df["technology_name"] == x["technology"])
         ]["cooling_fraction"]
-        x[col] = x[col] * cooling_fraction
 
-    results: list[Any] = []
+        # Log unmatched rows
+        if cooling_fraction.empty:
+            log.warning(
+                f"No cooling_fraction found for node_loc: {col2}, technology: {x['technology']}"
```

Comment:
Ruff says this line is too long and it doesn't presume to split up long string correctly, so please make this a two-line string manually to pass the code quality check.